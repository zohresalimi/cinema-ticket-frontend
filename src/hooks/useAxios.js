import { useEffect, useReducer, useMemo, useCallback } from "react";
import axios from "axios";
import qs from "qs";

const baseUrl = process.env.REACT_APP_BASE_URL;
const actions = {
  REQUEST_START: "REQUEST_START",
  REQUEST_END: "REQUEST_END",
};

async function fetchData({ path, ...config }, dispatch) {
  try {
    dispatch({ type: actions.REQUEST_START });
    const result = await axios({
      ...config,
      url: `${baseUrl}${path}`,
      paramsSerializer: (params) => {
        return qs.stringify(params);
      },
    });
    dispatch({ type: actions.REQUEST_END, payload: result.data });
  } catch (err) {
    dispatch({
      type: actions.REQUEST_END,
      payload: err,
      error: true,
    });
  }
}

const useAxios = (path, config) => {
  function reducer(state, action) {
    switch (action.type) {
      case actions.REQUEST_START:
        return {
          ...state,
          response: null,
          loading: true,
          error: null,
        };
      case actions.REQUEST_END:
        return {
          ...state,
          loading: false,
          ...(action.error ? {} : { data: action.payload.data }),
          [action.error ? "error" : "response"]: action.payload,
        };
      default:
        return state;
    }
  }

  function createInitialState({ manual }) {
    return {
      loading: !manual,
      error: null,
      response: null,
    };
  }

  const options = useMemo(
    () => ({
      data: {},
      method: "get",
      manual: false,
      path,
      ...config,
    }),
    [config, path]
  );
  const [state, dispatch] = useReducer(reducer, createInitialState(options));

  useEffect(() => {
    if (!options.manual) {
      fetchData(options, dispatch);
    }
  }, [options, dispatch, path]);

  const refetch = useCallback(
    (payload) => {
      return fetchData({ ...options, ...payload }, dispatch);
    },
    [options, dispatch]
  );

  return [state, refetch];
};

export default useAxios;
