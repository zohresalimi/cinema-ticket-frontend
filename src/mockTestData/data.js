import React from "react";
import { Router } from "@reach/router";
import AppContext from "../store/context";
import Theme from "../Styles/Theme";

export const getTestStore = () => {
  return {
    dispatch: () => {},
    state: {
      user: {
        name: "",
        email: "",
      },
      movies: {
        children: [
          {
            _id: "5fa71823adc463397a50c501",
            actors: ["Holland Roden", "Ronen Rubinstein", "Keegan Allen"],
            age: "7",
            coverImage:
              "https://catalog.cinema-api.com/cf/images/ncg-images/68bfeef55ada4e7ab0606db2af0f40ad.jpg?width=240&version=EA2E80EE161E5DA161C6E26487054C9F&format=webp",
            description:
              "Poppy and Kvist embark on an adventure that will take them far beyond",
            director: "aaron woodley",
            duration: "1:31",
            genre: ["family", "children"],
            images: [],
            largeImage:
              "https://catalog.cinema-api.com/cf/images/ncg-images/79276cead82341d2a004f8e8f99a88e9.jpg?width=1920&version=08F0C83177D59FB94C4C809F63F67E2A&format=webp",
            name: "trolls 2: world tour",
            originalTitle: "swedish",
            premiere: "2020-10-03T15:18:52.722Z",
            rooms: ["5fa653d0a7f1201999dd1ac0"],
            trailerUrl: "https://youtu.be/P9A6hOg9QQ0",
          },
        ],
        premiered: [
          {
            _id: "5fa71823adc463397a50c502",
            actors: ["Holland Roden", "Ronen Rubinstein", "Keegan Allen"],
            age: "7",
            coverImage:
              "https://catalog.cinema-api.com/cf/images/ncg-images/68bfeef55ada4e7ab0606db2af0f40ad.jpg?width=240&version=EA2E80EE161E5DA161C6E26487054C9F&format=webp",
            description:
              "Poppy and Kvist embark on an adventure that will take them far beyond",
            director: "aaron woodley",
            duration: "1:31",
            genre: ["family", "children"],
            images: [],
            largeImage:
              "https://catalog.cinema-api.com/cf/images/ncg-images/79276cead82341d2a004f8e8f99a88e9.jpg?width=1920&version=08F0C83177D59FB94C4C809F63F67E2A&format=webp",
            name: "follow me",
            originalTitle: "swedish",
            premiere: "2020-10-05T15:18:52.722Z",
            rooms: ["5fa653c0643c06586bba7d78"],
            trailerUrl: "https://youtu.be/P9A6hOg9QQ0",
          },
        ],
        upcoming: [
          {
            _id: "5fa71823adc463397a50c503",
            actors: ["Holland Roden", "Ronen Rubinstein", "Keegan Allen"],
            age: "7",
            coverImage:
              "https://catalog.cinema-api.com/cf/images/ncg-images/68bfeef55ada4e7ab0606db2af0f40ad.jpg?width=240&version=EA2E80EE161E5DA161C6E26487054C9F&format=webp",
            description:
              "Poppy and Kvist embark on an adventure that will take them far beyond",
            director: "aaron woodley",
            duration: "1:31",
            genre: ["family", "children"],
            images: [],
            largeImage:
              "https://catalog.cinema-api.com/cf/images/ncg-images/79276cead82341d2a004f8e8f99a88e9.jpg?width=1920&version=08F0C83177D59FB94C4C809F63F67E2A&format=webp",
            name: "honest thief",
            originalTitle: "swedish",
            premiere: "2021-01-03T15:18:52.722Z",
            rooms: ["5fa653c0643c06586bba7d78"],
            trailerUrl: "https://youtu.be/P9A6hOg9QQ0",
          },
        ],
        allMovies: {
          "5fa71823adc463397a50c501": "children",
          "5fa71823adc463397a50c502": "premiered",
          "5fa71823adc463397a50c503": "upcoming",
        },
      },
      cinemas: [
        {
          _id: "5fa64fe234fbbf9945f3a3ce",
          name: "filmstaden kista",
          purchaseEndTime: "21:30",
          purchaseStartTime: "8:00",
          rooms: ["5fa653c0643c06586bba7d78", "5fa653d0a7f1201999dd1ac0"],
        },
      ],
      ticket: {
        movie: {},
        status: false,
        quantity: 0,
        price: 0,
        seatNumbers: new Map(),
        user: {},
        showing: {},
      },
      showings: [
        {
          _id: "5fbe9bd31f58462b5c759b5c",
          capacity: 4,
          cinema: "5fa64fe234fbbf9945f3a3ce",
          endTime: null,
          movie: {
            _id: "5fa71823adc463397a50c502",
            actors: ["Holland Roden", "Ronen Rubinstein", "Keegan Allen"],
            age: "7",
            description:
              "Poppy and Kvist embark on an adventure that will take them far beyond",
            director: "aaron woodley",
            duration: "1:31",
            genre: ["family", "children"],
            images: [],
            name: "follow me",
            originalTitle: "swedish",
            premiere: "2020-10-05T15:18:52.722Z",
            rooms: ["5fa653c0643c06586bba7d78"],
            trailerUrl: "https://youtu.be/P9A6hOg9QQ0",
          },
          price: 100,
          room: { _id: "5fa653c0643c06586bba7d78" },

          seats: [
            [
              { seatNumber: 1, taken: "available" },
              { seatNumber: 1, taken: "available" },
            ],
            [
              { seatNumber: 1, taken: "available" },
              { seatNumber: 1, taken: "available" },
            ],
          ],
          startTime: "2020-11-13T16:00:00.000Z",
        },
      ],
    },
  };
};

export const WithProvider = (props) => {
  const { state, dispatch } = getTestStore();
  return (
    <Theme>
      <AppContext.Provider
        value={{ state: props.defaultValue || state, dispatch }}
      >
        <Router>{props.children}</Router>
      </AppContext.Provider>
    </Theme>
  );
};
