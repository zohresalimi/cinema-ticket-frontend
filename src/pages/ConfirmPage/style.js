import styled from "styled-components";

const Wrapper = styled.div`
  font-family: ${(props) => props.theme.fontFamily.sf};
  width: 100%;
  margin-top: 40px;
  font-size: ${(props) => props.theme.fontSizes.content};
  color: ${(props) => props.theme.colors.textColor};
  margin-bottom: 20px;
  h4 {
    font-size: ${(props) => props.theme.fontSizes.medium};
  }
  p {
    font-size: ${(props) => props.theme.fontSizes.content};
    margin: 5px 0;
    span {
      font-size: ${(props) => props.theme.fontSizes.medium};
      margin: 0 10px;
    }
  }
  .sub-title {
    font-size: ${(props) => props.theme.fontSizes.small};
    color: ${(props) => props.theme.colors.gray};
    margin: 0 10px;
  }
  .movie-info {
    display: flex;
    justify-content: end;
    .info {
      display: flex;
      flex-direction: column;
      margin-left: 20px;
    }
  }
  .row-box {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    margin: 10px 5px;
    border-bottom: 1px solid #ccc;
  }

  @media only screen and (max-width: 375px) {
    .movie-info{
      display: flex;
      flex-direction: column;
    }

`;

export { Wrapper as default };
