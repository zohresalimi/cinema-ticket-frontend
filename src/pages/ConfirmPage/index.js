/* eslint-disable react-hooks/exhaustive-deps */
import React, {
  useContext,
  useState,
  useMemo,
  useEffect,
  useCallback,
} from "react";
import { Link, useLocation, useNavigate } from "@reach/router";
import { useTranslation } from "react-i18next";
import Moment from "react-moment";
import { loadStripe } from "@stripe/stripe-js";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faSpinner,
  faAngleLeft,
  faEnvelope,
} from "@fortawesome/free-solid-svg-icons";
import useAxios from "../../hooks/useAxios";
import AppContext from "../../store/context";
import formatPrice from "../../utils/formatPrice";

import {
  Container,
  Row,
  Button,
  Breadcrumb,
  Alert,
  Input,
} from "../../Styles/StyleComponents";
import Wrapper from "./style";

const stripePromise = loadStripe(process.env.REACT_APP_PUBLISHABLE_KEY);

function ConfirmPage() {
  const { state } = useContext(AppContext);
  const { t, i18n } = useTranslation();
  const { state: locationState } = useLocation();
  const navigate = useNavigate();
  const { cinema, movie, price, quantity, seatNumbers, showing } = state.ticket;
  const [email, setEmail] = useState();
  const [emailError, setEmailError] = useState(null);
  const [checked, setChecked] = useState(false);
  const seatsArray = Array.from(seatNumbers).map((seat) => {
    return [seat[0], [...seat[1]].sort((a, b) => a - b)];
  });

  const [
    { error: ticketError, response: ticketResponse, loading: ticketLoading },
    createTicket,
  ] = useAxios("/api/v1/tickets", {
    manual: true,
    data: {
      email,
      quantity,
      seatNumbers: seatsArray,
      price,
      unitAmount: showing.price,
      movieName: movie.name,
      roomName: showing.room.name,
      cinemaName: cinema.name,
      showing: showing._id,
      movieCover: movie.coverImage,
    },
    method: "post",
  });

  const [
    { response: checkoutResponse, loading: checkoutInProgress },
    createCheckout,
  ] = useAxios("/api/v1/checkout/create", {
    manual: true,
    method: "post",
  });

  useEffect(() => {
    async function createSession() {
      if (ticketResponse && !checkoutInProgress && !checkoutResponse) {
        await createCheckout({
          data: {
            ticketId: ticketResponse.data._id,
          },
        });
      }
    }

    createSession();
  }, [checkoutInProgress, ticketResponse, checkoutResponse]);

  useEffect(() => {
    async function goToCheckout() {
      if (checkoutResponse) {
        const stripe = await stripePromise;
        const result = await stripe.redirectToCheckout({
          sessionId: checkoutResponse.id,
        });
        if (result.error) {
          navigate("order/cancel");
        }
      }
    }

    goToCheckout();
  }, [checkoutResponse]);

  const emailValidation = useCallback(() => {
    const pattern = new RegExp(
      /^(("[\w-\s]+")|([\w-]+(?:\.[\w-]+)*)|("[\w-\s]+")([\w-]+(?:\.[\w-]+)*))(@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$)|(@\[?((25[0-5]\.|2[0-4][0-9]\.|1[0-9]{2}\.|[0-9]{1,2}\.))((25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\.){2}(25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\]?$)/i
    );

    if (pattern.test(email) === false && email) {
      setEmailError("please enter a valid email");
    } else setEmailError(null);
  }, [email]);

  const enableButton = useMemo(() => email && checked && !emailError, [
    email,
    checked,
    emailError,
  ]);

  return (
    <Container>
      <Row>
        <div className="full-width">
          <Breadcrumb>
            <Link
              to="../seat-selection"
              state={{ category: locationState.category }}
            >
              <FontAwesomeIcon icon={faAngleLeft} className="mr-10" />
              {t("seat selection")}
            </Link>
            <Link to="/">{t("sing in")}</Link>
          </Breadcrumb>
          <Wrapper>
            <h2>{t("reservation")}</h2>
            <div className="movie-info">
              <img src={movie.coverImage} alt="" />
              <div className="info">
                <h4>{t(movie.name)}</h4>
                <p>
                  <Moment
                    locale={i18n.language}
                    calendar={showing.startTime}
                    format="dddd D MMMM"
                    tz="Europe/Stockholm"
                  />
                </p>
                <p>
                  <Moment
                    locale={i18n.language}
                    date={showing.startTime}
                    format="HH:mm"
                    tz="Europe/Stockholm"
                  />
                </p>
                <p>{cinema.name}</p>
                <p>{t(showing.room.name)}</p>
                <div>
                  {seatsArray &&
                    seatsArray.map(([row, seats]) => (
                      <div key={`r${row}`}>
                        <span className="sub-title">{t("row")}</span>
                        <span>{row + 1}</span>
                        <span className="sub-title">{t("seat number")}</span>
                        <span>{row + 1}</span>
                        <span>{seats.map((seat) => seat + 1).join(", ")}</span>
                      </div>
                    ))}
                </div>
              </div>
            </div>
            <div className="row-box">
              <p>{`${quantity} ticket`}</p>
              <p>{formatPrice(showing.price)}</p>
            </div>
            <div className="row-box">
              <p>{t("to pay")}</p>
              <p>{formatPrice(price)}</p>
            </div>
            <div>
              <label>
                {t("your e-mail")}
                <Input>
                  <div className="input-group">
                    <div className="input-group-text">
                      <FontAwesomeIcon icon={faEnvelope} />
                    </div>
                  </div>
                  <input
                    onInput={(e) => setEmail(e.target.value)}
                    onBlur={() => emailValidation()}
                    type="text"
                    name="name"
                    placeholder="example@gmail.com"
                  />
                </Input>
                {emailError && <Alert>{t(emailError)}</Alert>}
              </label>
            </div>
            <div>
              <label>
                <input
                  name="checkbox"
                  type="checkbox"
                  className="m-10"
                  onChange={() => setChecked(!checked)}
                />
                {t("i accept the  terms of purchase.")}
              </label>
            </div>
            <Button role="link" disabled={!enableButton} onClick={createTicket}>
              {ticketLoading && <FontAwesomeIcon icon={faSpinner} spin />}
              {t("continue")}
            </Button>
            {ticketError && <Alert>{t("try again")}</Alert>}
          </Wrapper>
        </div>
      </Row>
    </Container>
  );
}

export default ConfirmPage;
