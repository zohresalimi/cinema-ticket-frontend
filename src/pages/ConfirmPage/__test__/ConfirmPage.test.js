import * as React from "react";
import * as ReachRouter from "@reach/router";

import { act, render } from "@testing-library/react";
import { when } from "jest-when";
import "@testing-library/jest-dom/extend-expect";

import * as hooks from "../../../hooks/useAxios";
import { getTestStore, WithProvider } from "../../../mockTestData/data";
import ConfirmPage from "../";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({
    t: (k) => k,
    i18n: {
      language: "sv",
      changeLanguage: () => new Promise(() => {}),
    },
  }),
}));

async function renderWrapper(props) {
  let component;

  act(() => {
    component = render(
      <WithProvider>
        <ConfirmPage {...props} path="/" />
      </WithProvider>
    );
  });

  return component;
}

describe.skip("Confirm Page Component Testing", () => {
  afterAll(() => {
    jest.clearAllMocks();
  });

  test("take snapshot", async () => {
    const mockUseLocation = jest.fn(() => ({
      state: {
        category: "premiered",
      },
    }));
    jest.spyOn(ReachRouter, "useLocation").mockImplementation(mockUseLocation);
    const testState = getTestStore();
    const seatsMap = new Map();
    seatsMap.set(1, [1, 2]);
    seatsMap.set(2, [1, 4]);
    testState.ticket = {
      ...testState.ticket,
      cinema: testState.cinemas[0],
      seatNumbers: seatsMap,
      showing: {
        ...testState.showings[0],
        room: {
          name: "room name",
        },
      },
    };
    const mockDispatch = jest.fn();
    const mockUseContext = jest.fn(() => ({
      dispatch: mockDispatch,
      state: testState,
    }));
    jest.spyOn(React, "useContext").mockImplementation(mockUseContext);
    const { container } = await renderWrapper();
    expect(container.firstChild).toMatchSnapshot();
  });

  test("should call checkout api", async () => {
    const mockUseLocation = jest.fn(() => ({
      state: {
        category: "premiered",
      },
    }));
    jest.spyOn(ReachRouter, "useLocation").mockImplementation(mockUseLocation);
    const testState = getTestStore();
    const seatsMap = new Map();
    seatsMap.set(1, [1, 2]);
    seatsMap.set(2, [1, 4]);
    testState.ticket = {
      ...testState.ticket,
      cinema: testState.cinemas[0],
      seatNumbers: seatsMap,
      showing: {
        ...testState.showings[0],
        room: {
          name: "room name",
        },
      },
    };
    const mockDispatch = jest.fn();
    const mockUseContext = jest.fn(() => ({
      dispatch: mockDispatch,
      state: testState,
    }));
    jest.spyOn(React, "useContext").mockImplementation(mockUseContext);
    const mockUseAxios = jest.fn();
    when(mockUseAxios)
      .calledWith("/api/v1/checkout/create", {
        manual: true,
        method: "post",
      })
      .mockReturnValue([
        {
          response: {
            id: "test_id",
          },
          loading: true,
        },
      ]);

    when(mockUseAxios).calledWith("/api/v1/tickets", {
      manual: true,
      data: {
        email,
        quantity,
        seatNumbers: seatsMap,
        price,
        unitAmount: showing.price,
        movieName: movie.name,
        roomName: showing.room.name,
        cinemaName: cinema.name,
        showing: showing._id,
        movieCover: movie.coverImage,
      },
      method: "post",
    });

    jest.spyOn(hooks, "default").mockImplementation(mockUseAxios);
    const { container } = await renderWrapper();
    expect(container.firstChild).toMatchSnapshot();
  });
});
