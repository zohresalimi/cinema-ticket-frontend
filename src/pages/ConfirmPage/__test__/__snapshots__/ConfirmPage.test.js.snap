// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm Page Component Testing take snapshot 1`] = `
<div
  style="outline: none;"
  tabindex="-1"
>
  <div
    class="StyleComponents__Container-sc-11erdxt-0 fFgwWO"
  >
    <div
      class="StyleComponents__Row-sc-11erdxt-1 jhoSVF"
    >
      <div
        class="full-width"
      >
        <h2
          class="StyleComponents__Breadcrumb-sc-11erdxt-4 cNjPuO"
        >
          <a
            href="/seat-selection"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-angle-left fa-w-8 mr-10"
              data-icon="angle-left"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 256 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
                fill="currentColor"
              />
            </svg>
            seat selection
          </a>
          <a
            aria-current="page"
            href="/"
          >
            sing in
          </a>
        </h2>
        <div
          class="style__Wrapper-sc-183f9xs-0 lovBnU"
        >
          <h2>
            reservation
          </h2>
          <div
            class="movie-info"
          >
            <img
              alt=""
            />
            <div
              class="info"
            >
              <h4 />
              <p>
                <time
                  datetime="1606695508693"
                >
                  m√•ndag 30 november
                </time>
              </p>
              <p>
                <time
                  datetime="1605283200000"
                >
                  17:00
                </time>
              </p>
              <p>
                filmstaden kista
              </p>
              <p>
                room name
              </p>
              <div>
                <div>
                  <span
                    class="sub-title"
                  >
                    row
                  </span>
                  <span>
                    2
                  </span>
                  <span
                    class="sub-title"
                  >
                    seat number
                  </span>
                  <span>
                    2
                  </span>
                  <span>
                    2, 3
                  </span>
                </div>
                <div>
                  <span
                    class="sub-title"
                  >
                    row
                  </span>
                  <span>
                    3
                  </span>
                  <span
                    class="sub-title"
                  >
                    seat number
                  </span>
                  <span>
                    3
                  </span>
                  <span>
                    2, 5
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="row-box"
          >
            <p>
              0 ticket
            </p>
            <p>
              100,00 kr
            </p>
          </div>
          <div
            class="row-box"
          >
            <p>
              to pay
            </p>
            <p>
              0,00 kr
            </p>
          </div>
          <div>
            <label>
              your e-mail
              <div
                class="StyleComponents__Input-sc-11erdxt-6 jWtKJs"
              >
                <div
                  class="input-group"
                >
                  <div
                    class="input-group-text"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-envelope fa-w-16 "
                      data-icon="envelope"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                </div>
                <input
                  name="name"
                  placeholder="example@gmail.com"
                  type="text"
                />
              </div>
            </label>
          </div>
          <div>
            <label>
              <input
                class="m-10"
                name="checkbox"
                type="checkbox"
              />
              i accept the  terms of purchase.
            </label>
          </div>
          <button
            class="StyleComponents__Button-sc-11erdxt-3 fWbzoT"
            disabled=""
            role="link"
          >
            continue
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
